src/Components/Aspire.Kafka.Consumer/AspireKafkaConsumerExtensions.cs

```diff
@@ -0,0 +1,87 @@
+/// </summary>
+public static class AspireKafkaConsumerExtensions
+{
+    private const string DefaultConfigSectionName = "Aspire:Kafka:Consumer";
+
+    /// <summary>
+    /// Registers <see cref="IConsumer{TKey,TValue}"/> as a singleton in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>    /// 
+    /// <param name="configureConsumerBuilder">A method used for customizing the <see cref="ConsumerBuilder{TKey,TValue}"/>.</param>
+    /// <param name="configureConsumerConfig">An optional method that can be used for customizing the <see cref="ConsumerConfig"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:Kafka:Consumer" section.</remarks>
+    public static void AddKafkaConsumer<TKey, TValue>(this IHostApplicationBuilder builder, string connectionName, Action<ConsumerBuilder<TKey, TValue>>? configureConsumerBuilder = null, Action<ConsumerConfig>? configureConsumerConfig = null)
+        => AddKafkaConsumer(builder, DefaultConfigSectionName, configureConsumerBuilder, configureConsumerConfig, connectionName, serviceKey: null);
+
+    /// <summary>
+    /// Registers <see cref="IConsumer{TKey,TValue}"/> as a keyed singleton for the given <paramref name="name"/> in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="name">The name of the component, which is used as the <see cref="ServiceDescriptor.ServiceKey"/> of the service and also to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureConsumerBuilder">A method used for customizing the <see cref="ConsumerBuilder{TKey,TValue}"/>.</param>
+    /// <param name="configureConsumerConfig">An optional method that can be used for customizing the <see cref="ConsumerConfig"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:Kafka:Consumer:{name}" section.</remarks>
+    public static void AddKeyedKafkaConsumer<TKey, TValue>(this IHostApplicationBuilder builder, string name, Action<ConsumerBuilder<TKey, TValue>>? configureConsumerBuilder = null, Action<ConsumerConfig>? configureConsumerConfig = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        AddKafkaConsumer(builder, $"{DefaultConfigSectionName}:{name}", configureConsumerBuilder, configureConsumerConfig, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddKafkaConsumer<TKey, TValue>(
```

Comment:
Any health checks, telemetry etc?

Reply:
Not yet. I'm thinking to pull logs and metrics from librdkafka using the below delegates that can be registered within `ProducerBuilder<TKey, TValue>` and `ConsumerBuilder<TKey, TValue>` [SetLogHandler](https://docs.confluent.io/platform/current/clients/confluent-kafka-dotnet/_site/api/Confluent.Kafka.ConsumerBuilder-2.html#Confluent_Kafka_ConsumerBuilder_2_SetLogHandler_System_Action_Confluent_Kafka_IConsumer__0__1__Confluent_Kafka_LogMessage__), [SetErrorHandler](https://docs.confluent.io/platform/current/clients/confluent-kafka-dotnet/_site/api/Confluent.Kafka.ConsumerBuilder-2.html#Confluent_Kafka_ConsumerBuilder_2_SetErrorHandler_System_Action_Confluent_Kafka_IConsumer__0__1__Confluent_Kafka_Error__), [SetStatisticsHandler](https://docs.confluent.io/platform/current/clients/confluent-kafka-dotnet/_site/api/Confluent.Kafka.ConsumerBuilder-2.html#Confluent_Kafka_ConsumerBuilder_2_SetStatisticsHandler_System_Action_Confluent_Kafka_IConsumer__0__1__System_String__). The [STATISTICS.json](https://github.com/confluentinc/librdkafka/blob/master/STATISTICS.md) contains metrics.